From 2c0f62135cdd2f6472f01da7af0ef65f825481e7 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Fri, 13 Oct 2023 16:13:48 +0530
Subject: [PATCH 101/101] Adding Error Message for Invalid-MTUSize

Description:
   Added Condition and Error message for check the MTUSize value is between minimum and maximum Value.

1) Patching a value lesser than device support
patch call
https://x.x.x.x/redfish/v1/Managers/bmc/EthernetInterfaces/eth0

Body
{
 "MTUSize":67
}

Response
{
    "error": {
        "@Message.ExtendedInfo": [
            {
                "@odata.type": "#Message.v1_1_1.Message",
                "Message": "The value '\"67\"' for the property MTUSize is not in the supported range of acceptable values.",
                "MessageArgs": [
                    "\"67\"",
                    "MTUSize"
                ],
                "MessageId": "Base.1.16.0.PropertyValueOutOfRange",
                "MessageSeverity": "Warning",
                "Resolution": "Correct the value for the property in the request body and resubmit the request if the operation failed."
            }
        ],
        "code": "Base.1.16.0.PropertyValueOutOfRange",
        "message": "The value '\"67\"' for the property MTUSize is not in the supported range of acceptable values."
    }
}

2)Patching a value greater than device support
patch call
https://x.x.x.x/redfish/v1/Managers/bmc/EthernetInterfaces/eth0

Body
{
 "MTUSize":1501
}

Response
{
    "error": {
        "@Message.ExtendedInfo": [
            {
                "@odata.type": "#Message.v1_1_1.Message",
                "Message": "The value '\"1501\"' for the property MTUSize is not in the supported range of acceptable values.",
                "MessageArgs": [
                    "\"1501\"",
                    "MTUSize"
                ],
                "MessageId": "Base.1.16.0.PropertyValueOutOfRange",
                "MessageSeverity": "Warning",
                "Resolution": "Correct the value for the property in the request body and resubmit the request if the operation failed."
            }
        ],
        "code": "Base.1.16.0.PropertyValueOutOfRange",
        "message": "The value '\"1501\"' for the property MTUSize is not in the supported range of acceptable values."
    }
}
Signed-off-by: prasannas <prasannas@ami.com>
---
 redfish-core/lib/ethernet.hpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index aa4ea79..bc88f15 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -38,6 +38,9 @@
 #include <string_view>
 #include <vector>
 
+#define MAX_MTU 1500
+#define MIN_MTU 68
+
 namespace redfish
 {
 
@@ -957,6 +960,14 @@ inline void
 {
     sdbusplus::message::object_path objPath = "/xyz/openbmc_project/network/" +
                                               ifaceId;
+    if ((mtuSize < MIN_MTU) || (mtuSize > MAX_MTU))
+    {
+        std::string mtu = std::to_string(mtuSize);
+        std::string_view mtuview(mtu);
+        messages::propertyValueOutOfRange(asyncResp->res, mtuview, "MTUSize");
+        return;
+    }
+
     sdbusplus::asio::setProperty(
         *crow::connections::systemBus, "xyz.openbmc_project.Network", objPath,
         "xyz.openbmc_project.Network.EthernetInterface", "MTU", mtuSize,
-- 
2.9.5

